src/utils/apollo-client/cache.ts

```diff
@@ -0,0 +1,17 @@
+import { InMemoryCache } from 'apollo-boost';
+
+export const appCache = new InMemoryCache();
+
+export interface authCache {
```

Comment:
1. I think interfaces should start  from capital letter - **AuthCache**. Not sure if there's `eslint` rule for that 🤔

2. Also I've noticed that field `__typename: 'auth';` is never changed and is repeated in many places in different files [error-link.ts](https://github.com/BuildIt-Poland/buddy-app/blob/713a315138b87a3354e145078bf47707184fd098/src/utils/apollo-client/error-link.ts), [AuthContext.ts](https://github.com/BuildIt-Poland/buddy-app/blob/713a315138b87a3354e145078bf47707184fd098/src/contexts/AuthContext.tsx#L79) and [cache.ts](https://github.com/BuildIt-Poland/buddy-app/pull/51/files#diff-9be50270c72fab22b3ecf83389d74290) - which I think could simplified by using  **class** as a type for **AuthCache**  with  `__typename: 'auth';` being initialized inside constructor as a default property and re-used everywhere you need to use instance of **AuthCache**.

```
  auth: {
    __typename: 'auth',
    tokenHasExpired: false,
  },
```


**cache.ts**
```
import { InMemoryCache } from 'apollo-boost';

export const appCache = new InMemoryCache();

type Auth = {
  __typename?: string;
  tokenHasExpired: boolean;
};

export class AuthCache {
  auth: Auth;
  constructor(auth: Auth) {
    this.auth = {
      __typename: 'auth',
      tokenHasExpired: auth.tokenHasExpired || false,
    };
  }
}

export const defaultCache = new AuthCache({ tokenHasExpired: false });
```

So instead of repeating the same object with only one field being changed

**error-link.ts**
```
            data: {
              auth: {
                __typename: 'auth',
                tokenHasExpired: true,
              },
            },
```
you can do like below: 

```
data: new AuthCache({ tokenHasExpired: false }),
```

Reply:
@michaelmoney I had different approach just made a pure function to accomplish that.

Reply:
Now I see it. Looks good 👍